src/plugins/intel_cpu/src/nodes/dft.cpp

```diff
@@ -404,24 +481,221 @@ void DFT::naiveDFT(float* data, size_t dataLength) const {
+    static int cacheSizeL3 = dnnl::utils::get_cache_size(3, false);
+    static int elementsPerCacheLine = cacheSizeL3 / sizeof(float);
+    size_t nComplex = dataLength / 2;
+
+    auto blockIteration =
+        [&](const size_t block, const size_t numBlocks, const size_t blockSize, const size_t nextIterationBlockSize) {
+            auto arg = jit_args_fft();
+
+            arg.src = inBuffer + block * nextIterationBlockSize * 2;
+            arg.dst = outBuffer + block * nextIterationBlockSize;
+            arg.twiddles = &twiddlesFFT[(numBlocks + block - 1) * 2];
+            arg.num_blocks = numBlocks;
+            arg.work_amount = nextIterationBlockSize;
+            arg.n_complex = nComplex;
+
+            (*fftKernel)(&arg);
+        };
+
+    size_t blockSize;
+    size_t nextIterationBlockSize = dataLength;
+    for (size_t numBlocks = 1; numBlocks < nComplex; numBlocks *= 2) {
+        blockSize = nextIterationBlockSize;
+        nextIterationBlockSize /= 2;
+        if (parallelize && blockSize >= 4 * elementsPerCacheLine) {
+            parallel_for(numBlocks, [&](const size_t block) {
+                blockIteration(block, 1, blockSize, nextIterationBlockSize);
+            });
+        } else {
+            blockIteration(0, numBlocks, blockSize, nextIterationBlockSize);
+        }
+
+        std::swap(inBuffer, outBuffer);
+    }
+
+    if (inverse) {
+        for (int64_t k = 0; k < dataLength; k++) {
+            inBuffer[k] /= nComplex;
+        }
+    }
+
+    return inBuffer;
+}
+
+void DFT::DFTJitExecutor::naiveDFT(float* data, size_t dataLength, bool inverse) const {
+    std::vector<float> outputBuffer(dataLength);
+    const size_t nComplex = dataLength / 2;
+    const auto& twiddles = twiddlesMapDFT.find(nComplex)->second;
+
+    parallel_for(nComplex, [&](size_t k) {
+        auto arg = jit_args_dft();
+
+        arg.src = data;
+        arg.dst = outputBuffer.data() + 2 * k;
+        arg.twiddles = twiddles.data() + 2 * k * nComplex;
+        arg.work_amount = nComplex;
+        arg.index = k;
+
+        (*dftKernel)(&arg);
+    });
+    cpu_memcpy(data, outputBuffer.data(), dataLength * sizeof(float));
+}
+
 bool DFT::created() const {
     return getType() == Type::DFT;
 }
 
-void DFT::createPrimitive() {}
+bool DFT::needPrepareParams() const {
+    return lastInverse != inverse;
+}
+
+void DFT::prepareParams() {
+    DFTKey key = {};
+
+    key.nodeAttrs.inverse = inverse;
+    key.nodeAttrs.hasDFT = false;
+    key.nodeAttrs.hasFFT = false;
+
+    axes = getAxes();
+    const auto outputShape = getChildEdgesAtPort(0)[0]->getMemory().getStaticDims();
+
+    for (size_t axis : axes) {
+        size_t nComplex = outputShape[axis];
+        if (!IsPowerOfTwo(nComplex)) {
+            key.nodeAttrs.hasDFT = true;
+        } else {
+            key.nodeAttrs.hasFFT = true;
+        }
+    }
+
+    auto buildExecutor = [&](const DFTKey& key) -> std::shared_ptr<DFTExecutor> {
```

Comment:
As I understand, both kernels are shape agnostic i.e. they do not depend on shapes at compilation time. So they is not required to be cached and can be compiled before inference in function *createPrimitive()*

Reply:
More precisely - their creation depends on shape size, kernels by itself are shape agnostic

Reply:
Ok, I see, but it is easy to join these kernels and pass one argument on execution.
Kernel recompilation significantly affects performance, thus need to create shape agnostic kernels. Cache algorithm was created as work around for dynamic shapes and should be removed in future.

Reply:
Done; also, I removed kernel's 'inverse' parameter, so it should be never recompiled now.